api/src/backend/tasks/jobs/integrations.py

```diff
@@ -0,0 +1,123 @@
+) -> tuple[bool, S3 | Connection]:
+    """
+    Create and return a boto3 S3 client using AWS credentials from an integration.
+
+    Args:
+        integration (Integration): The integration to get the S3 client from.
+
+    Returns:
+        tuple[bool, S3 | Connection]: A tuple containing a boolean indicating if the connection was successful and the S3 client or connection object.
+    """
+    session = AwsProvider(**integration.credentials).session.current_session
+    s3 = S3(
+        session=session,
+        bucket_name=integration.configuration["bucket_name"],
+        output_directory=integration.configuration["output_directory"],
+    )
+    connection = s3.test_connection(
+        session=session,
+        bucket_name=integration.configuration["bucket_name"],
+    )
+
+    if connection.is_connected:
+        return True, s3
+
+    return False, connection
+
+
+def upload_s3_integration(tenant_id: str, provider_id: str, file_path: str) -> bool:
+    """
+    Upload the specified output files to an S3 bucket from an integration.
+    If the S3 bucket environment variables are not configured,
+    the function returns None without performing an upload.
+
+    Args:
+        tenant_id (str): The tenant identifier, used as part of the S3 key prefix.
+        provider_id (str): The provider identifier, used as part of the S3 key prefix.
+        file_path (str): The local file system path to the output files to be uploaded.
+
+    Returns:
+        bool: True if all integrations were executed, False otherwise.
+
+    Raises:
+        botocore.exceptions.ClientError: If the upload attempt to S3 fails for any reason.
+    """
+    with rls_transaction(tenant_id):
+        integrations = list(
+            Integration.objects.filter(
+                integrationproviderrelationship__provider_id=provider_id,
+                integration_type=Integration.IntegrationChoices.S3,
+            )
+        )
+
+    if not integrations:
+        logger.error(f"No S3 integrations found for provider {provider_id}")
+        return False
+
+    integration_executions = 0
+    for integration in integrations:
+        integration_configuration = integration.configuration
+        integration_bucket_name = integration_configuration.get("bucket_name")
+        integration_output_directory = integration_configuration.get("output_directory")
+
+        try:
+            connected, s3 = get_s3_client_from_integration(integration)
+        except Exception as e:
+            logger.error(f"S3 connection failed for integration {integration.id}: {e}")
+            continue
+
+        if connected:
+            try:
+                for filename in os.listdir(file_path):
+                    local_path = os.path.join(file_path, filename)
+                    if not os.path.isfile(local_path):
+                        continue
+                    file_key = f"{integration_output_directory}/{filename}"
+                    s3.upload_file(
```

Comment:
Why are you not using SDK's `send_to_bucket`? We should aim to have all this login in the SDK and to adapt whatever is missing for the API to use it. If not you will need to replicate all the logic in there to handle content-type and file paths.

Reply:
This method was created because the API uses a completely different system to manage outputs. While we could replicate what’s in the CLI, it’s not the most efficient approach for the API. So instead of introducing changes just to mirror the CLI, a new function was created. Regarding the content-type handling, in my tests, I haven’t had to worry about it, files are uploaded correctly, the type is set properly, and everything can be viewed and downloaded without any issues.

We can discuss this in more detail and reach an agreement since it’s an important decision

Reply:
I get that. I think we should first analyse the SDK's status prior starting the work and make all the required changes. 

Regarding the `content-type`, we need to set it for each file because AWS sets only the file type.